File: boj/10773.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   10773.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jaeelee <jaeelee@student.42seoul.kr>       +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/03/10 09:26:49 by jaeelee           #+#    #+#             */
/*   Updated: 2022/03/10 09:36:21 by jaeelee          ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <stack>
using namespace std;

int main()
{
	int K;
	int n;
	int answer = 0;
	stack<int> s;

	cin >> K;

	for (int i = 0; i < K; i++)
	{
		cin >> n;
		if (n == 0)
			s.pop();
		else
			s.push(n);
	}

	while (!s.empty())
	{
		answer += s.top();
		s.pop();
	}
	cout << answer;

	return answer;
}

File: boj/10773.java
/*
   stak을 사용하는 문제
   0이 들어오면 pop, 그 외의 숫자들은 push를 시키고 남은 숫자들을 모두 더한다.
*/
import java.util.Scanner;
import java.util.Stack;

class Main {
    public static void main(String[] args) 
    {         
        Stack<Integer> s = new Stack<>(); //int형 스택 선언
        
        Scanner sc = new Scanner(System.in);
        int k = sc.nextInt();

        int n;
        for (int i = 0; i < k; i++)
        {
            n = sc.nextInt();
            
            if (n == 0)
                s.pop();
            else
                s.push(n);
        }
        
        int answer = 0;
        while (!s.empty())
            answer += s.pop();
        
        System.out.println(answer);       
    } 
}
